### ISSUE TITLE
🚀 - `Hint` - support 12 directional positioning

### ISSUE THREAD

**Issue by hakimio** [Tinkoff/taiga-ui#739]
# 🐞 Bug report

### Description

Right now [TuiDirection](https://github.com/TinkoffCreditSystems/taiga-ui/blob/173213fa013c73ea38e2afbc16c5ba3f3996bae3/projects/core/types/direction.ts#L7) is declared the following way:
```ts
export type TuiDirection = TuiHorizontalDirection | TuiSideDirection;
```
but I would expect it to also include `TuiVerticalDirection` as well:
```ts
export type TuiDirection = TuiHorizontalDirection | TuiVerticalDirection | TuiSideDirection;
```

**Comment by waterplea** [comment]
By design system you were not supposed to have those types of hints, but it makes sense to add them, I think. It should ultimately be 12 positions like ng-zorro-ant have:
![image](https://user-images.githubusercontent.com/11832552/134315672-a0e8eb63-a9e9-4dbd-8415-d742ea79ee2b.png)

Let's put this for 3.0 and rename directions like you suggested in other issue along with it.

**Comment by waterplea** [comment]
Fixed in 3.0
